ouroboros-consensus-shelley-test/test/Test/ThreadNet/RealTPraos.hs

```diff
@@ -14,25 +15,33 @@ import           Test.Tasty.QuickCheck
 
 import           Cardano.Crypto.Hash (ShortHash)
 import qualified Cardano.Crypto.KES.Class as KES
+import           Cardano.Slotting.EpochInfo.Impl (fixedSizeEpochInfo)
 
 import           Ouroboros.Consensus.Block
 import           Ouroboros.Consensus.Config.SecurityParam
+import           Ouroboros.Consensus.Ledger.Abstract
 import           Ouroboros.Consensus.Ledger.SupportsMempool (extractTxs)
 import           Ouroboros.Consensus.Node.NetworkProtocolVersion
 import           Ouroboros.Consensus.Node.ProtocolInfo
 import           Ouroboros.Consensus.NodeId
+import           Ouroboros.Consensus.Ticked
 
 import           Test.ThreadNet.General
+import           Test.ThreadNet.Network (TestNodeInitialization (..),
+                     nodeOutputFinalLedger)
 import           Test.ThreadNet.Infra.Shelley
 
 import           Test.Util.HardFork.Future (singleEraFuture)
 import           Test.Util.Orphans.Arbitrary ()
+import           Test.Util.Slots (NumSlots (..))
 
 import qualified Shelley.Spec.Ledger.BaseTypes as SL
 import qualified Shelley.Spec.Ledger.OCert as SL
+import qualified Shelley.Spec.Ledger.LedgerState as SL
 import qualified Shelley.Spec.Ledger.PParams as SL
 
 import           Ouroboros.Consensus.Shelley.Ledger (ShelleyBlock)
+import qualified Ouroboros.Consensus.Shelley.Ledger as SO
```

Comment:
`SO`? I have used `Shelley` as the qualifier for this import, but maybe adding the two extra imports to the import list is less redundant.

Reply:
Done, thanks.